Sources/Html/Node.swift

```diff
@@ -16,6 +19,29 @@ public enum Node {
   case text(String)
 }
 
+extension Node {
+  public var isEmpty: Bool {
+    switch self {
+    case let .comment(string), let .doctype(string), let .raw(string), let .text(string):
+      return string.isEmpty
+    case .element:
+      return false
+    case let .fragment(children):
+      return children.isEmpty || children.allSatisfy { $0.isEmpty }
```

Comment:
you can drop the `children.isEmpty` since `allSatisfy` returns true for empty collections.